### PR TITLE
Add Node#Send keys to standardize usage across drivers

### PR DESCRIPTION
**Pull request by twalpole**
Implementation of item discussed in Issue #1459  -  @jferris @jonleighton @mhoran Thoughts on implementing this in your drivers for Capybara 2.5 ?

### CODE REVIEW

File: lib/capybara/node/element.rb [hunk 1]
```diff
@@ -148,6 +148,87 @@ def right_click
...
+      # @param [String, Symbol, Array]
+      #
+      # Examples:
+      #
+      #     element.send_keys "foo"                     #=> value: 'foo'
+      #     element.send_keys "tet", :arrow_left, "s"   #=> value: 'test'
+      #     element.send_keys [:control, 'a'], :space   #=> value: ' '
+      #
+      # Symbols supported for keys
+      # :null
```
Comment (abotalov) [inline lib/capybara/node/element.rb]: trailing space

**Comment by mhoran** [comment]
There are a few keys that we can't send in capybara-webkit

```ruby
:left_shift
:left_control
:left_alt
:arrow_left
:arrow_up
:arrow_right
:arrow_down
:separator
:decimal
:command
```

There is no differentiation between shift and left shift, control and left control, or alt and left alt in Qt. We only have left, up, right, and down mappings, but nothing corresponding to "arrow".

Is `:separator` `|`? Is `:decimal` `.`?

Command is not a cross-platform key, so it is not supported by Qt. On OS X, `Qt::Key_Control` maps to command, and `Qt::Key_Meta` maps to control. We could conceivably map these as appropriate on each individual platform, but that may be surprising for users.

Finally, what should `:null` send?

**Comment by twalpole** [comment]
@mhoran - The list of symbols is what selenium understands
:left_shift, :left_alt, :left_control are just aliases for :shift, :alt, :passport_control: 
:arrow_up, :arrow_down, :arrow_right, :arrow_left  are aliases for :up, :down, :right, :left.    
:command is an alias for :meta.  
:decimal is the numeric keypad .  
:add, :divide, :subtract, :multiply are the numeric keypad instances of +, /, -, *

:separator is the numeric keypad "seperator" key (different from the :decimal key) - the keycode is 108 and I'm guessing its a key returned by non-english keyboards although I can't find info that specifically states that.

:null - appears to be used in lower level key routines in selenium to trigger release of all modifier keys.  It is probably not needed for this so I'm removing it from the list

**Comment by jferris** [comment]
> There is no differentiation between shift and left shift, control and left control, or alt and left alt in Qt. We only have left, up, right, and down mappings, but nothing corresponding to "arrow".

I'm guessing that if we just sent shift, control, and alt, that would be least surprising to users. Same goes for up/arrow_up.

**Comment by twalpole** [comment]
@jferris I have removed the arrow_x and left_x aliases from the list of required supported key symbols - For future info DOM Level 3 does support identifying left vs right instances of shift, alt, control via the location attribute of a KeyboardEvent however not all browsers support it at the moment https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent.location and selenium doesn't currently provide a way to set it, so we may have to revisit left_shift, right_shift, etc in the future

**Pull request merged**